OBAKit/Location/OBARegionStorage.m

```diff
@@ -84,12 +90,12 @@ - (void)writeRegionsToDisk:(NSArray<OBARegionV2*>*)regions {
         return nil;
     }
 
-    NSArray<OBARegionV2*>* regions = nil;
+    NSData *objectData = [NSData dataWithContentsOfFile:OBARegionStorage.localRegionsFilePath];
+    NSError *error;
```

Comment:
Sorry I didn't catch this before, but many APIs will fill their error parameters with garbage when the API returns successfully. A safer check is:

    NSError *error = nil;
    NSArray *regions = [NSKeyedUnarchiver unarchiveTopLevelObjectFor:objectData error:&error];
    if (!regions) {
         DDLogError(@"Failed to unarchive persisted regions: %@", error);
     }